sys/include/net/gnrc/netdev2.h

```diff
@@ -81,8 +96,81 @@ typedef struct gnrc_netdev2 {
      * @brief PID of this adapter for netapi messages
      */
     kernel_pid_t pid;
+
+#ifdef MODULE_GNRC_MAC
+    uint16_t mac_info;  /**< general information for the MAC protocol */
+#endif
 } gnrc_netdev2_t;
 
+#ifdef MODULE_GNRC_MAC
+
+/**
+ * @brief get the 'rx_started' state of the device
+ *
+ * This function checks whether the device has started receiving a packet.
+ *
+ * @param[in] dev  ptr to netdev2 device
+ *
+ * @return         the rx_started state
+ */
+static inline bool gnrc_netdev2_get_rx_started(gnrc_netdev2_t *dev)
+{
+    return (dev->mac_info & GNRC_NETDEV2_MAC_INFO_RX_STARTED);
+}
+
+/**
+ * @brief set the rx_started state of the device
+ *
+ * This function is intended to be called only in the _event_cb function of
+ * the netdev2 device.
```

Comment:
>  @daniel-k was the rx_started boolean from your implementation supposed to only be called in the event callback?

In my implementation `rx_started` was part of the global LwMAC state struct. I didn't follow the refactoring to that detail, so I don't know how introduced `gnrc_netdev2_set_rx_started()`, sorry :/

But this function might be dangerous in general. If I remember correctly, `_event_cb()` can be called in interrupt context as well as in thread context, but it is definetly not thread-safe. Furthermore, manipulating the rx_started information from outside the MAC layer will also cause malfunction.